src/Symfony/Component/VarDumper/Dumper/HtmlDumper.php

```diff
@@ -267,6 +357,136 @@ function isCtrlKey(e) {
+    
+                    addEventListener(searchInput, 'keydown', function (e) {
+                        /* Don't intercept escape key in order to not start a search */
+                        if (27 === e.keyCode) {
+                            return;
+                        }
+    
+                        delay(function () {
+                            state.reset();
+                            collapseAll(root);
+                            resetHighlightedNodes(root);
+                            var searchQuery = e.target.value;
+                            if ('' === searchQuery) {
+                                counter.textContent = '0 on 0';
+    
+                                return;
+                            }
+    
+                            var xpathResult = doc.evaluate('//pre[@id="' + root.id + '"]//span[@class="sf-dump-str" or @class="sf-dump-key" or @class="sf-dump-public" or @class="sf-dump-protected" or @class="sf-dump-private"][contains(child::text(), \"' + searchQuery + '\")]', document, null, XPathResult.ORDERED_NODE_ITERATOR_TYPE, null);
+    
+                            while (node = xpathResult.iterateNext()) {
+                                state.nodes.push(node);
+                            }
+    
+                            var currentNode = state.current();
+                            if (currentNode) {
+                                reveal(currentNode);
+                                highlight(root, currentNode, state.nodes);
+                            }
+    
+                            counter.textContent = (state.isEmpty() ? 0 : state.idx + 1) + ' on ' + state.count();
+                        }, 400);
+                    });
+                    Array.from(search.querySelectorAll('.sf-dump-search-input-next, .sf-dump-search-input-previous')).forEach(function (btn) {
+                        addEventListener(btn, 'click', function (e) {
+                            e.preventDefault();
+                            var direction = -1 !== e.target.className.indexOf('next') ? 'next' : 'previous';
+                            'next' === direction ? state.next() : state.previous();
+                            searchInput.focus();
+                            collapseAll(root);
+                            var currentNode = state.current();
+                            if (currentNode) {
+                                reveal(currentNode);
+                                highlight(root, currentNode, state.nodes);
+                            }
+    
+                            counter.textContent = (state.isEmpty() ? 0 : state.idx + 1) + ' on ' + state.count();
+                        })
+                    });
+                }
+    
+                root.querySelector('.sf-dump-search-input').focus();
+            } else if (27 === e.keyCode) {
+                /* ESC key */
+                e.preventDefault();
```

Comment:
no, it prevents it when search *has been focused*, because you don't remove the listener after unfocusing the search, and so any future ESC key will still be prevented by this listener.